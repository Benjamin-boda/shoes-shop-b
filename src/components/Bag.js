import React, { useEffect, useState } from "react";
import {useSelector, useDispatch} from "react-redux";
import {startRemoveProduct, startEditProduct, startResetProducts} from "../actions/products";
import {startAddOrder} from "../actions/orders";
import products from "../tests/fixtures/products";

export const Bag = () => {
    const [sizeUpdate, setSizeUpdate] = useState("")
    const [productUpdate, setProductUpdate] = useState("")
    const productsInStore  = useSelector(state => state.products)
    const dispatch = useDispatch()

    const sizes = [37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47]
    

    const sizeEdit = (e) => {
        setSizeUpdate(e.target.value)
    }

    useEffect(() => {
        if (productUpdate !== "" & productsInStore.lenght > 0) {
            const filterProductArray = productsInStore.filter((product) => productUpdate === product.name)
            const productEdit = {
                image : {
                    default : filterProductArray[0].image.default
                },
                name : filterProductArray[0].name,
                size : sizeUpdate,
                price : filterProductArray[0].price,
                boughtAt : filterProductArray[0].boughtAt
            }
            const id = filterProductArray[0].id
            dispatch(startEditProduct(id, productEdit))    
        }
    }, [sizeUpdate])

    const paymentOrder = () => {
        productsInStore.map((product) => {
            product.boughtAt = Date()
            dispatch(startAddOrder(product))})
        dispatch(startResetProducts())              
    }

    const sumPrice = productsInStore.length > 0 ? productsInStore.map(product => product.price).reduce((acc, price) => acc + price) : "0"

    return (
        <div className="bag__content">
            <div className="bag__box">
                {productsInStore.length > 0 ?
                    productsInStore.map((product) => 
                        <div className="bag" key={product.name} onMouseEnter={() => setProductUpdate(product.name)}>
                            <img className="bag__image" src={product.image.default}/>
                            <p data-testid={product.name} className="bag__name">{product.name}</p>
                            <select className="bag__select" name="size" onChange={sizeEdit}>
                                <option value={product.size}>{product.size}</option>
                                {sizes.map((size) => 
                                    <option value={size} key={size}>{size}</option>
                                    )}
                            </select>
                            <p className="bag__name">Price : {product.price} €</p>
                            <button className="bag__button" onClick={() => dispatch(startRemoveProduct(product))}><img className="bag__cross__button" src={require("../images/cross button.png").default}/></button>
                        </div>
                    ) : <p className="bag__empty">Nothing in the bag</p>
                }
            </div>
            
            <p className="bag__price">Total price : {sumPrice} €</p>

            <div className="bag__payment__div">
                <button className="bag__payment" onClick={paymentOrder} disabled={productsInStore.length === 0 ? true : false}>Payment</button>
            </div>
            
        </div>
    )
}