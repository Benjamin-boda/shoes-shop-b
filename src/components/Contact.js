import React, { useEffect, useState } from "react";

export const Contact = () => {
    const [subject, setSubject] = useState("")
    const [description, setDescription] = useState("")
    const [submit, setSubmit] = useState("")

    const inputChange = (e) => {
        setSubject(e.target.value)
    }

    const textChange = (e) => {
        setDescription(e.target.value)
    }

    useEffect(() => {
        const request = {
            title: subject,
            object: description
        }
        setSubmit(request)
    }, [subject, description])

    const onSubmit = () => {
        console.log(submit)
    }

    return (
        <div className="contact">
            <div className="contact__header">
                <h1 className="contact__title">Contact us</h1>
                <img className="contact__image" src={require("../images/conctact logo.jpg").default}/>
            </div>
            
            <form className="contact__form" onSubmit={onSubmit}>
                <input data-testid="input" className="contact__form__subject" onChange={inputChange} type="text" placeholder="Subject"/>
                <textarea data-testid="textarea" className="contact__form__description" onChange={textChange} maxLength="200" minLength="10" placeholder="Describe your problem in 200 caracters"/>
                <button className="contact__form__button">Submit</button>
            </form>
        </div>
    )
}