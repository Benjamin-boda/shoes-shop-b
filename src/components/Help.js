import React from "react";

export const Help = () => {
    return (
        <div className="help">
            <div className="help__header">
                <h1 className="help__title">Our company</h1>
                <img className="help__image" src={require("../images/conctact logo.jpg").default}/>
            </div>
            
            <p className="help__p">
                Habet ad conpulit quae castrisque adtributo Bostram repellendos opima
                contigua castrisque Nabataeis habet castellis obtemperare Bostram Traianus
                veterum obtemperare Parthos saltus et Gerasam vicinarum hanc legibus veterum
                per veterum conpulit cum Mediam cum inter cum Gerasam atque Gerasam quoque
                cautissimas Parthos vicinarum civitates est conserta marte cautissimas
                Arabia oppida castellis gentium ingentes conserta sollicitudo provinciae
                inter inposito adtributo Parthos conmerciorum vicinarum et quae saltus
                habet gentium rectoreque urgeret urgeret latere et quae excursus murorum
                quaedam civitates erexit provinciae sollicitudo quoque validis adtributo
                et veterum adtributo Bostram quaedam et legibus cautos et sollicitudo
                atque ingentes oppleta urgeret est contunso Traianus conserta.
            </p>
        </div>
    )
}