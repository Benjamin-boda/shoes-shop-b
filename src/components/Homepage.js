import React, {useState} from "react";
import {wallpapers} from "../products/wallpapers";

export const Homepage = () => {
    const [countHomepageImg, setcountHomepageImg] = useState(1)

    const nextHomepageImg = () => {
        if (countHomepageImg >= 1 & countHomepageImg < 3) {
            setcountHomepageImg(countHomepageImg + 1)
        }
    }

    const precedentHomepageImg = () => {
        if (countHomepageImg > 1 & countHomepageImg <= 3) {
            setcountHomepageImg(countHomepageImg - 1)
        }
    }

    return (
         <div className="homepage__overflow">
            {wallpapers.map((wallpaper) => 
                countHomepageImg === wallpaper.number ? 
                    <div key={wallpaper.number}>
                        <img className="homepage" src={wallpaper.img.default}/>
                        <p className="homepage__text">{wallpaper.description}</p>
                    </div> : undefined
                )}
            <button className="homepage__next" onClick={nextHomepageImg}><img className="homepage__next__image" src={require("../images/left arrow.png").default}/></button>
            <button className="homepage__precedent" onClick={precedentHomepageImg}><img className="homepage__precedent__image" src={require("../images/left arrow.png").default}/></button>
        </div>
    )
   
}