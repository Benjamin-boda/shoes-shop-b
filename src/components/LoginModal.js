import React from "react";
import Modal from "react-modal";

export const LoginModal = ({modalIsOpen, setModalIsOpen}) => {

    return (
        <Modal
            isOpen = {modalIsOpen}
            onRequestClose = {() => setModalIsOpen(false)}
            contentLabel = "Login Modal"
            closeTimeoutMS = {200}
            className="loginmodal"
            ariaHideApp={false}
        >
            <h3 className="loginmodal__title">You need to be connected to see our products</h3>
            <button className="loginmodal__button" onClick={() => setModalIsOpen(false)}>Close</button>
        </Modal>
    )
}