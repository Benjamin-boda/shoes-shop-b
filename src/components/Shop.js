import React, {useState} from "react";
import {Link} from "react-router-dom";
import {products} from "../products/products";

export const Shop = () => {
    const [currentPage, setCurrentPage] = useState(1)
    const numberPage = Math.ceil(products.length/4)
    let pages = []

    for (let i = 1; i < numberPage + 1; i++) {
        pages.push(i)
    }

    return (
        <div>
            <div className="shop">
                {products.slice((currentPage - 1) * 4, currentPage * 4)
                    .map((product, index) =>
                    <Link className={`shop__shoes shop__shoes__${index}`} to={`/product/${product.number}`} key={product.number}>
                        <img className="shop__image" src={product.img.default}/>
                        <p className="shop__name">{product.name}</p>
                    </Link>
                )}
            </div>
            
            <div className="shop__pages">
                {pages.map((page) => 
                    <Link className="shop__pages__link" key={page}  onClick={() => setCurrentPage(page)} to={`/shop/${page}`}>
                        {page}
                    </Link>
                    )}
            </div>
            
        </div> 
    )
}