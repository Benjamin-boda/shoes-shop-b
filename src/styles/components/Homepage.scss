.homepage {
    height: 86vh;
    width: 100%;
    z-index: 0;
}

.homepage__overflow {
    height: 85vh;
    overflow: hidden;
    width: 100%;
}

.homepage__text {
    color: white;
    font-family: "MonserratBlack";
    font-size: $font-size-medium;
    padding-bottom: $s-size;
    position: relative;
    right: -5vh;
    top: -20vh;
    z-index: 10;

    @media (min-width: $desktop-breakpoint) {
        font-size: $font-size-large;
    }
}

.homepage__precedent {
    background: none;
    border: none;
    left: -5rem;
    position: relative;
    text-decoration: none;
    top: -73vh;
    width: 4rem;
    z-index: 10;
}

.homepage__next {
    background: none;
    border: none;
    left: 87vw;
    position: relative;
    text-decoration: none;
    top: -73vh;
    width: 4rem;
    z-index: 10;

    @media (min-width: $desktop-breakpoint) {
        left: 94vw;
    }

    @media (min-width: $desktop-breakpoint-large) {
        left: 97vw;
    }
}

.homepage__precedent__image {
    width: 4rem;
}

.homepage__next__image {
    transform: rotate(180deg);
    width: 4rem;
}