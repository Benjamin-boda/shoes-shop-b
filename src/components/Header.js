import React, {useState ,useContext} from "react";
import {Link} from "react-router-dom";
import {useDispatch, useSelector} from "react-redux";
import {startLogin, startLogout} from "../actions/auth";
import {AuthContext} from "../firebase/Auth";
import {LoginModal} from "./LoginModal";

export const Header = () => {
    const {currentUser} = useContext(AuthContext);
    const [accountHovering, setAccountHovering] = useState(false)
    const [modalIsOpen, setModalIsOpen] = useState(false)
    const productsInStore = useSelector(state => state.products)
    const dispatch = useDispatch()

    const openModal = () => {
        if (!!currentUser) {
            setModalIsOpen(false)
        } else {
            setModalIsOpen(true)
        }
    }

    return (
        <div className="header">
            <img className="header__logo" src={require("../images/logo.png").default}/>
            <Link className="header__link" to="/">Home</Link>
            <Link className="header__link" to={!!currentUser ? "/shop/1" : ""} onClick={openModal}>Shop</Link>
            <Link className="header__link" to="/contact">Contact</Link>
            <Link className="header__link" to="/help">Help</Link>
            
            <div className="header__account">
                <Link 
                    data-testid="account"
                    className="header__link__account"
                    to=""
                    onClick={() => !!currentUser ? undefined : dispatch(startLogin())}
                    onMouseEnter={() => !!currentUser ? setAccountHovering(true) : setAccountHovering(false)}
                >
                    {!!currentUser ? "User account" : "Login"}
                </Link>
                {accountHovering ? 
                    <ul className="header__dropdown" onClick={() => setAccountHovering(false)} onMouseLeave={() => setAccountHovering(false)}>
                        <Link className="header__link__account__dropdown" to="/orders">
                            <li className="header__dropdown__item">My account</li>
                        </Link>
                        <li className="header__dropdown__item" onClick={() => dispatch(startLogout())}>Logout</li>
                        <li className="header__dropdown__item" onClick={() => setAccountHovering(false)}>Close</li>
                    </ul> : undefined
                }
            </div>
            
            <div>
                <Link to={!!currentUser ? "/bag" : ""} onClick={openModal}><img className="header__bag" src={require("../images/bag.png").default}/></Link>
                <p className="header__bag__counter">{productsInStore.length > 0 ? productsInStore.length : 0}</p>
            </div>
            

            <LoginModal modalIsOpen={modalIsOpen} setModalIsOpen={setModalIsOpen}/>
        </div>
    )
}