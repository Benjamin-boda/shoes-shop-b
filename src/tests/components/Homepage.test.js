import React from "react";
import {Homepage} from "../../components/Homepage";
import {render, cleanup} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect';

afterEach(cleanup);

const { asFragment } = render(<Homepage />)

test("should match the snapshot", () => {
    
    expect(asFragment(<Homepage/>)).toMatchSnapshot()
})

test("should contain this text", () => {
    const { getByText } = render(<Homepage />)

    expect(getByText("Check out our new products")).toBeTruthy()
})