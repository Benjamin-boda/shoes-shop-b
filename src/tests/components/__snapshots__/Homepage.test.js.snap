// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="homepage__overflow"
  >
    <div>
      <img
        class="homepage"
      />
      <p
        class="homepage__text"
      >
        Check out our new products
      </p>
    </div>
    <button
      class="homepage__next"
    >
      <img
        class="homepage__next__image"
      />
    </button>
    <button
      class="homepage__precedent"
    >
      <img
        class="homepage__precedent__image"
      />
    </button>
  </div>
</DocumentFragment>
`;
